halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Utils.java

```diff
@@ -105,6 +105,61 @@ static private boolean exists(KubernetesAccount account, String namespace, Strin
     }
   }
 
+  static public boolean isReady(KubernetesAccount account, String namespace, String service) {
+    log.info("Checking readiness for " + service);
+    List<String> command = kubectlPrefix(account);
+
+    if (StringUtils.isNotEmpty(namespace)) {
+      command.add("-n=" + namespace);
+    }
+
+    command.add("get");
+    command.add("po");
+
+    command.add("-l=cluster=" + service);
+    command.add("-o=jsonpath='{.items[*].status.containerStatuses[*].ready}'");
+    // This command returns a space-separated string of true/false values indicating whether each of
+    // the pod's containers are READY.
+    // e.g., if we are querying two spin-orca pods and both pods' monitoring-daemon containers are
+    // READY but the orca containers are not READY, the output may be 'true false true false'.
+
+    JobRequest request = new JobRequest().setTokenizedCommand(command);
+
+    String jobId = DaemonTaskHandler.getJobExecutor().startJob(request);
+
+    JobStatus status;
+    try {
+      status = DaemonTaskHandler.getJobExecutor().backoffWait(jobId);
+    } catch (InterruptedException e) {
+      throw new DaemonTaskInterrupted(e);
+    }
+
+    if (status.getState() != JobStatus.State.COMPLETED) {
+      throw new HalException(Problem.Severity.FATAL, String.join("\n",
+          "Unterminated readiness check for " + service + " in " + namespace,
+          status.getStdErr(),
+          status.getStdOut()));
+    }
+
+    if (status.getResult() == JobStatus.Result.SUCCESS) {
```

Comment:
I think this can fail if no containers could be found -- which might happen immediately after a deployment.

Reply:
Upon experimentation I found that if no containers exist the status result will still be SUCCESS, with stdout containing two single quotes ('').